### PR TITLE
Use window.devicePixelRatio to sharpen canvas images

### PR DESCRIPTION
**Pull request by stronglink**
Using `devicePixelRatio` allows us to improve the visual quality of canvas renderings by accounting for the ratio between physical and virtual pixels. Most notably, canvas charts will look much better on Apple's Retina Displays.

**Before**
screen shot 2017-09-11 at 3 00 58 pm

**After**
screen shot 2017-09-11 at 3 00 51 pm

### CODE REVIEW

File: src/plot/series/canvas-wrapper.js [hunk 1]
```diff
@@ -104,8 +104,13 @@ function buildLayers(newChildren, oldChildren) {
   });
 }
 class CanvasWrapper extends Component {
+
   componentDidMount() {
-    this.drawChildren(this.props, null, this.refs.canvas.getContext('2d'));
+    const ctx = this.refs.canvas.getContext('2d');
+    const { pixelRatio } = this.props;
```
Comment (mcnuttandrew) [inline src/plot/series/canvas-wrapper.js]: I think your build is failing bc it expects `const {pixelRatio} = this.props;`

File: src/plot/xy-plot.js [hunk 2]
```diff
@@ -413,7 +413,8 @@ class XYPlot extends React.Component {
       marginBottom,
       marginRight,
       innerHeight,
-      innerWidth
+      innerWidth,
+      pixelRatio: window.devicePixelRatio
```
Comment (mcnuttandrew) [inline src/plot/xy-plot.js]: might be good to generalize this for node env testing, eg make a default at the top that is something like

```javascript
const DEFAULT_PIXEL_RATIO = window && window.devicePixelRatio || 1;
```

also, would you mind alphabetizing this list?

Comment (stronglink) [reply src/plot/xy-plot.js]: Sounds like a good idea. I've also alphabetized the list.

File: src/plot/series/canvas-wrapper.js (hunk 1 above)
Comment (stronglink) [reply src/plot/series/canvas-wrapper.js]: Yea, that's the cause. I've modified the README to point out `npm test` doesn't lint the code.

File: src/plot/xy-plot.js [hunk 3]
```diff
@@ -56,6 +56,8 @@ const DEFAULT_MARGINS = {
   bottom: 40
 };
 
+const DEFAULT_PIXEL_RATIO = window && window.devicePixelRatio || 1;
```
Comment (mcnuttandrew) [inline src/plot/xy-plot.js]: hmm it'd probably be better if this was a default prop in the canvas-wrapper component? sorry to flip around on this

Comment (stronglink) [reply src/plot/xy-plot.js]: No problem! I've changed it.

**Comment by mcnuttandrew** [comment]
👍👍👍

**Pull request merged**